Israel National Newspaper Library.js

```diff
@@ -0,0 +1,143 @@
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.includes("/article/") || url.includes("/page/")) {
+		return "newspaperArticle";
+	}
+	return false;
+}
+
+async function doWeb(doc, url) {
+	await scrape(doc, url);
+}
+
+function detectLanguageFromText(text) {
+	if (/[\u0590-\u05FF]/.test(text)) return "he"; // Hebrew range
+	if (/[\u0600-\u06FF]/.test(text)) return "ar"; // Arabic range
+	if (/[a-zA-Z]/.test(text)) return "en"; // Basic Latin letters
+	return null;
+}
+
+async function scrape(doc, url) {
+	const item = new Zotero.Item("newspaperArticle");
+
+	// Get JSON-LD data
+	let jsonLDData = null;
+	const jsonLD = doc.querySelector('script[type="application/ld+json"]');
+	if (jsonLD) {
+		try {
+			jsonLDData = JSON.parse(jsonLD.textContent);
+			const headline = ZU.trimInternal(jsonLDData.headline);
+			if (headline) item.title = headline;
+		} catch (e) {
+			ZU.debug("Error parsing JSON-LD for newspaper: " + e);
+		}
+	}
+
+	// Persistent link
+	const linkNode = doc.querySelector("#sectionleveltabpersistentlinkarea .persistentlinkurl");
+	item.url = linkNode ? ZU.trimInternal(linkNode.textContent) : url;
+
+	// Abstract + Full Text
+	const paragraphs = Array.from(doc.querySelectorAll("#pagesectionstextcontainer p"))
+		.map(p => ZU.trimInternal(p.textContent))
+		.filter(Boolean);
+
+	if (paragraphs.length) {
+		item.abstractNote = paragraphs[0];
+		
+		// Language detection
+		const sampleText = paragraphs.join(" ").slice(0, 1000);
+		const detectedLang = detectLanguageFromText(sampleText);
+		if (detectedLang) {
+			const langMap = { he: "heb", ar: "ara", en: "eng" };
+			item.language = langMap[detectedLang] || detectedLang;
+		}
+	}
+
+	// Date from breadcrumb
+	const dateNode = doc.querySelector('li.breadcrumb-item:nth-child(3)');
+	if (dateNode) {
+		item.date = ZU.trimInternal(dateNode.textContent);
+	}
+
+	// Get publication from NLI script data
+	const nliScript = doc.querySelector('script#nlijs');
+	if (nliScript) {
+		const rawJSON = nliScript.getAttribute('data-nli-data-json');
+		if (rawJSON) {
+			try {
+				const json = JSON.parse(rawJSON.replace(/&quot;/g, '"').replace(/&amp;/g, '&'));
+				if (json.publicationTitle) {
+					item.publicationTitle = ZU.trimInternal(json.publicationTitle);
+				}
+			} catch (e) {
+				ZU.debug("Failed to parse data-nli-data-json: " + e);
+			}
+		}
+	}
+
+	// Page number handling
+	if (url.includes("/page/")) {
+		const match = url.match(/\/page\/(\d+)(?:\/|$)/);
+		if (match) {
+			item.pages = match[1];
+		}
+		// Clean URL for citation
+		item.url = url.split('?')[0].split('#')[0];
```

Comment:
A persistent URL, extracted by an earlier section of code, would look like this: `https://www.nli.org.il/en/newspapers/?a=d&d=echo19341102-01`, so this line would rewrite it with an invalid one.

Reply:
I had something else in mind, but I now realize you're just overriding the `item.url` with a parsed value of `url` regardless of what `item.url`'s value was before, so the behavior I described won't happen, so never mind.